documentation/modules/ROOT/pages/configuration/topic-auto-create-config.adoc

```diff
@@ -0,0 +1,216 @@
+topic name matches the given regular expression.
+
+|3
+|Then we define the configuration for the `applicationlogs` group.
+
+|4
+|`topic.creation.applicationlogs.include` defines a regular expression to match all topics that start
+with `dbserver1.logs.applog-`. The config defined for the `applicationlogs` group will only be
+applied when the topic name matches the given regular expression. As there's also the `exclude`
+property defined at position *<5>* all topics matching this `include` regular expression might be
+further restricted by the that `exlude` property.
+
+|5
+|`topic.creation.applicationlogs.exclude` defines a regular expression to match all topics that start
+with `dbserver1.logs.applog-old-`. The config defined for the `applicationlogs` group will only be
+applied when the topic name does *not* match the given regular expression. As there's also the
+`include` property set for this group the `applicationlogs` group will only be applied to topics
+which name matches the `include` regular expression/s *and* _not_ match the `exclude` regular
+expression/s.
+|===
+
+=== Registering Custom Groups
+
+Finally, we need to register the two defined custom groups `inventory` and `applicationlogs` with
+the `topic.creation.groups` property:
+
+[source,options="nowrap",json]
+----
+{
+    ...
+
+    "topic.creation.groups": "inventory,applicationlogs",
+
+     ...
+}
+----
+
+A complete connector JSON config will look like that:
+
+[source,options="nowrap",json]
+----
+{
+    ...
+
+    "topic.creation.default.replication.factor": 3,
+    "topic.creation.default.partitions": 10,
+    "topic.creation.default.cleanup.policy": "compact",
+    "topic.creation.default.compression.type": "lz4"
+    "topic.creation.groups": "inventory,applicationlogs",
+    "topic.creation.inventory.include": "dbserver1\\.inventory\\.*",
+    "topic.creation.inventory.replication.factor": 3,
+    "topic.creation.inventory.partitions": 20,
+    "topic.creation.inventory.cleanup.policy": "compact",
+    "topic.creation.inventory.delete.retention.ms": 7776000000,
+    "topic.creation.applicationlogs.include": "dbserver1\\.logs\\.applog-.*",
+    "topic.creation.applicationlogs.exclude": "dbserver1\\.logs\\.applog-old-.*",
+    "topic.creation.applicationlogs.replication.factor": 1,
+    "topic.creation.applicationlogs.partitions": 20,
+    "topic.creation.applicationlogs.cleanup.policy": "delete",
+    "topic.creation.applicationlogs.retention.ms": 7776000000,
+    "topic.creation.applicationlogs.compression.type": "lz4"
+}
+----
+
+ifdef::community[]
+== Additional resources
+- Debezium Blog: https://debezium.io/blog/2020/09/15/debezium-auto-create-topics/[Auto-creating Debezium Change Data Topics]
+- Kafka Improvement Proposal about adding topic auto-creation to Kafka Connect: https://cwiki.apache.org/confluence/display/KAFKA/KIP-158%3A+Kafka+Connect+should+allow+source+connectors+to+set+topic-specific+settings+for+new+topics[KIP-158]
+endif::community[]
```

Comment:
I only allowed these resources in community. Is that fine or can this be in the downstream docs as well?

Reply:
No, we typically don't link to blogs/upstream from prod docs.

Reply:
So yes, it's fine then?

Reply:
Yeah :)